code/modules/admin/verbs/SDQL2/SDQL_spells/spell_edit_menu.dm

```diff
@@ -0,0 +1,925 @@
+	)
+
+	var/static/list/special_var_lists = list(
+		"projectile_type" = "projectile_var_overrides",
+		"hand_path" = "hand_var_overrides",
+	)
+
+	var/static/list/enum_vars = list(
+		"invocation_type" = list("none", "whisper", "shout", "emote"),
+		"selection_type" = list("view", "range"),
+		"smoke_spread" = list(0, 1, 2, 3),
+		"random_target_priority" = list(0, 1),
+	)
+
+	var/static/list/static_data
+
+	//base64 representations of any icons that may need to be displayed
+	var/action_icon_base64
+	var/projectile_icon_base64
+	var/hand_icon_base64
+	var/overlay_icon_base64
+	var/mouse_icon_base64
+
+/datum/give_sdql_spell/New(user, target)
+	if(!CONFIG_GET(flag/sdql_spells))
+		to_chat(user, span_warning("SDQL spells are disabled."))
+		qdel(src)
+		return
+	src.user = CLIENT_FROM_VAR(user)
+
+	if(istype(target, /obj/effect/proc_holder/spell))
+		target_spell = target
+		var/mob/living/spell_owner = target_spell.owner.resolve()
+		if(spell_owner)
+			target_mob = spell_owner
+		else
+			to_chat(user, span_warning("[target_spell] does not have an owner, or its owner was qdelled. This REALLY shouldn't happen."))
+			qdel(src)
+			return
+	else if(isliving(target))
+		target_mob = target
+	else
+		to_chat(user, span_warning("Invalid target."))
+		qdel(src)
+		return
+	if(target_spell)
+		load_vars_from(target_spell)
+
+/datum/give_sdql_spell/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "SDQLSpellMenu", "Give SDQL Spell")
+		ui.open()
+		ui.set_autoupdate(FALSE)
+
+/datum/give_sdql_spell/ui_state(mob/user)
+	return GLOB.admin_state
+
+/datum/give_sdql_spell/ui_status(mob/user, datum/ui_state/state)
+	if(QDELETED(target_mob))
+		return UI_CLOSE
+	return ..()
+
+/datum/give_sdql_spell/ui_close(mob/user)
+	qdel(src)
+
+#define SANITIZE_NULLIFY 0
+#define SANITIZE_STRINGIFY 1
+
+/datum/give_sdql_spell/ui_data(mob/user, params)
+	var/list/data = list()
+	if(target_spell)
+		data["type"] = copytext("[target_spell.type]", 31, -5)
+		data["fixed_type"] = TRUE
+	else
+		data["type"] = spell_type
+		data["fixed_type"] = FALSE
+	data["saved_vars"] = saved_vars
+	data["list_vars"] = json_sanitize_list_vars(list_vars, SANITIZE_STRINGIFY)
+	if(parse_result)
+		data["parse_errors"] = parse_result["parse_errors"]
+		data["parsed_type"] = parse_result["type"]
+		data["parsed_vars"] = parse_result["vars"]
+		data["parsed_list_vars"] = json_sanitize_list_vars(parse_result["list_vars"], SANITIZE_STRINGIFY)
+	else
+		data["parse_errors"] = null
+		data["parsed_type"] = null
+		data["parsed_vars"] = null
+		data["parsed_list_vars"] = null
+	data["action_icon"] = action_icon_base64
+	data["projectile_icon"] = projectile_icon_base64
+	data["hand_icon"] = hand_icon_base64
+	data["overlay_icon"] = overlay_icon_base64
+	data["mouse_icon"] = mouse_icon_base64
+	data["alert"] = alert
+	alert = ""
+	return data
+
+/datum/give_sdql_spell/ui_static_data(mob/user)
+	if(!static_data)
+		static_data = list(
```

Comment:
Remove this redundant variable. Since you're returning this static data once, there is no need to cache it in variable. Cost of creating these lists is minimal.